src/agent/src/message_queue_utils.cpp

```diff
@@ -11,20 +11,23 @@ namespace
 
 boost::asio::awaitable<std::string> GetMessagesFromQueue(std::shared_ptr<IMultiTypeQueue> multiTypeQueue,
                                                          MessageType messageType,
-                                                         std::function<nlohmann::json()> getMetadataInfo)
+                                                         std::function<std::string()> getMetadataInfo)
 {
-    const auto message = co_await multiTypeQueue->getNextNAwaitable(messageType, NUM_EVENTS, "", "");
+    const auto messages = co_await multiTypeQueue->getNextNAwaitable(messageType, NUM_EVENTS, "", "");
 
-    nlohmann::json jsonObj;
     std::string output;
 
     if (getMetadataInfo != nullptr)
     {
-        jsonObj = getMetadataInfo();
-        output = jsonObj.dump() + "\n";
+        output = getMetadataInfo() + "\n";
     }
 
-    co_return output + message.metaData + "\n" + message.data.dump();
+    for (const auto& message : messages)
+    {
+        output += message.metaData + "\n" + message.data.dump() + "\n";
```

Comment:
In this way, we may avoid having a chain at the end of the chain.

```suggestion
        output += "\n" + message.metaData + "\n" + message.data.dump();
```

and deleting "\n" from

```
output = getMetadataInfo() + "\n"
```

Reply:
Done!